### PR TITLE
feat: support automatic certificates over Cert Manager

### PR DESCRIPTION
**Pull request by nettoclaudio**
This PR adds support to automatic certificates issuing and renewal. There are several providers to handle this task, I've started implementing it with Cert Manager, but we can change or even extend it in the future.

This PR also changes how we store certificates in Secrets. We used to create a new Secret for each change on certificates - we left some Secrets orphaned in this process. We only handle a single Secret for certificates and trigger a rollout on pods by updating its annotation with a SHA256 digest of those certificates.

The final user can enable this integration by issuing a command:

```bash
$ rpaasv2 certificates add --cert-manager --issuer lets-encrypt --dns app.example.com
```

and removing it:

```bash
$ rpaasv2 certificates remove --cert-manager
```

---
This change is [Reviewable](https://reviewable.io/reviews/tsuru/rpaas-operator/104)

### CODE REVIEW

File: internal/controllers/certificates/cert_manager.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+	certmanagerSecretName := cert.Spec.SecretName
+
+	if err = client.Delete(ctx, cert); err != nil {
+		return err
+	}
+
+	var s corev1.Secret
+	err = client.Get(ctx, types.NamespacedName{
+		Name:      certmanagerSecretName,
```
Comment (arthurcgc) [inline internal/controllers/certificates/cert_manager.go:68]: why don't you just use cert.Spec.SecretName? I see no gain from the attribution in line 60.

File: internal/controllers/certificates/cert_manager.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+func getCertManagerIssuer(ctx context.Context, client client.Client, instance *v1alpha1.RpaasInstance) (*cmmeta.ObjectReference, error) {
+	var issuer cmv1.Issuer
+
+	err := client.Get(ctx, types.NamespacedName{
+		Name:      instance.Spec.DynamicCertificates.CertManager.Issuer,
+		Namespace: instance.Namespace,
+	}, &issuer)
+
+	if err == nil {
```
Comment (arthurcgc) [inline internal/controllers/certificates/cert_manager.go:147]: same here

**Comment by arthurcgc** [comment]
very minor comments, but LGTM. Great work! @nettoclaudio

File: internal/controllers/certificates/auto_certificates.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+		return fmt.Errorf("rpaasinstance cannot be nil")
+	}
+
+	var s corev1.Secret
+	err := c.Get(ctx, types.NamespacedName{
+		Name:      secretNameForCertificates(instance),
+		Namespace: instance.Namespace,
+	}, &s)
+
+	if err != nil && k8serrors.IsNotFound(err) {
```
[the remark this answers is not in the document]
Comment (nettoclaudio) [reply internal/controllers/certificates/auto_certificates.go:62]: mmm... I don't know I've ever used this way :stuck_out_tongue: But for sure, I'll replace it

**Pull request merged**